### ISSUE TITLE
DRA module

### ISSUE THREAD

**Issue by lsh-med** [caijilia/ERDUnet#2]
class DRA(nn.Module):  # Differential Regional Attention
    def __init__(self, in_ch=64, class_num=2, value=0.5, simple=1, large=0):
        super(DRA, self).__init__()
        self.threshold = value
        self.s_mode = simple
        self.l_mode = large
        self.sigmoid = nn.Sigmoid()
        self.add_learn = nn.Sequential(
            nn.Conv2d(in_channels=in_ch, out_channels=in_ch, kernel_size=1),
            nn.BatchNorm2d(in_ch),
            nn.GELU()
        )
        self.to_class_1 = nn.Conv2d(in_channels=in_ch, out_channels=class_num, kernel_size=1)
        self.to_class_2 = nn.Conv2d(in_channels=in_ch, out_channels=class_num, kernel_size=1)
        self.region_learn = nn.Sequential(
            nn.Conv2d(in_channels=class_num, out_channels=1, kernel_size=1),
            nn.BatchNorm2d(1),
            nn.GELU()
        )

    def forward(self, x1, x2):
        b, c, h, w = x1.shape
        if self.s_mode == 1:
            x1_in = 1 * (x1 > self.threshold)
            x2_in = 1 * (x2 > self.threshold)
            edge_diff = torch.abs(x1_in - x2_in)
            x = (edge_diff) * x1 + x1
            x = self.add_learn(x)
        elif self.l_mode == 1:
            x1_in = self.to_class_1(x1)
            x2_in = self.to_class_2(x2)
            x1_in = 1 * (x1_in > self.threshold)
            x2_in = 1 * (x2_in > self.threshold)
            edge_diff = torch.abs(x1_in - x2_in) * 1.000000
            edge_diff = self.region_learn(edge_diff)
            x = self.sigmoid(edge_diff) * x1 + x1
            x = self.add_learn(x)
        return x
作者您好，我想问下 elif self.l_mode == 1:
            x1_in = self.to_class_1(x1)
            x2_in = self.to_class_2(x2) 将特征的通道数修改为class_num的具体意义是什么
